packages/suite/src/components/onboarding/OnboardingCancelButtonContext.tsx

```diff
@@ -0,0 +1,25 @@
+import { ReactNode, createContext, useContext, useState } from 'react';
+
+type CancelButtonContextData = {
+    onCancelHandler: (() => void) | null;
+    setOnCancelHandler: null | ((onCancelHandler: (() => void) | null) => void);
+};
+
+const CancelButtonContext = createContext<CancelButtonContextData>({
+    onCancelHandler: null,
+    setOnCancelHandler: null,
+});
+
+export function OnboardingCancelButtonContext({ children }: { children: ReactNode }) {
+    const [onCancelHandler, setOnCancelHandler] = useState<(() => void) | null>(null);
+
+    return (
+        <CancelButtonContext.Provider value={{ onCancelHandler, setOnCancelHandler }}>
+            {children}
+        </CancelButtonContext.Provider>
+    );
+}
+
+export function useOnboardingCancelButtonContext() {
+    return useContext(CancelButtonContext);
+}
```

Comment:
Nit arrow instead of named for brevity
```suggestion
export const useOnboardingCancelButtonContext = () => useContext(CancelButtonContext);
```